module1.js

```diff
@@ -54,6 +65,10 @@ function question3(number) {
 
 function question4(array) {
   // TODO:
+  var newArray = [];
+  for (i = 0; i < array.length; i + 2) {
+    return newArray.push(array[i]);
```

Comment:
As soon as the code reaches the `return` statement, it finishes running, so the loop only iterates once. You're not returning an array either - the `push()` method actually returns the new length of the array, which is 1, as soon as you add an element to it (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push)

Also, the `i + 2` doesn't really do anything - can you see why? Are you assigning the result of this addition anywhere? If you actually remove the `return` from the loop and run the code, the loop never actually stops. Figure out why ;)

Reply:
I think the answer would be this. However, not sure if the test is correct as the function question4 is not taking any array as an input.

Reply:
You're right, the test is wrong - I think it's been fixed since the quiz though so if you update your repo it should be correct :) However, Look at the expected array in the test - the element starts from 2, not from 0. The test still fails. You're very close though!